src/controllers/customFields.gs

```diff
@@ -0,0 +1,97 @@
+/** 
```

Comment:
Will be refactoring access to the fields in this customFields functionality so refactored this into a single controller as we agreed. I've added unit testing around the field access also.
Manually tested.